actions/workflows/promote_package.yaml

```diff
@@ -0,0 +1,100 @@
+---
+version: '2.0'
+
+st2cd.promote_package:
+    description: Promote a package to production.
+    type: direct
+    input:
+      - package
+      - version
+      - revision
+      - os
+      - repo
+    tasks:
+
+      init:
+        action: core.local
+        input:
+          cmd: "mkdir -p /tmp/st2-up/{ubuntu/trusty,debian/wheezy,debian/jessie,el/6,el/7}"
+        publish:
+          rpm: "<% $.package %>-<% $.version %>-<% $.revision %>.x86_64.rpm"
+          deb: "<% $.package %>_<% $.version %>-<% $.revision %>_amd64.deb"
+          source: "<% $.repo = 'enterprise' and 'enterprise-staging' or $.repo = 'enterprise-unstable' and 'enterprise-staging-unstable' or $.repo = 'stable' and 'staging-stable' or $.repo = 'unstable' and 'staging-unstable' %>"
+          token: ""
+        on-success:
+          - download_trusty: "<% $.os in list('all', 'deb', 'trusty') %>"
+          - download_wheezy: "<% $.os in list('all', 'deb', 'wheezy') %>"
+          - download_jessie: "<% $.os in list('all', 'deb', 'jessie') %>"
+          - download_el6: "<% $.os in list('all', 'rpm', 'el6') %>"
+          - download_el7: "<% $.os in list('all', 'rpm', 'el7') %>"
+
+      download_trusty:
+        action: core.local
+        input:
+          cmd: "wget -O /tmp/st2-up/ubuntu/trusty/<% $.deb %> https://packagecloud.io/StackStorm/<% $.source %>/ubuntu/pool/trusty/main/<% $.package.substring(0,1) %>/<% $.package %>/<% $.deb %>"
+        on-success: upload_trusty
+#        on-error: cleanup
+      upload_trusty:
+        action: core.local
+        input:
+          cmd: "LANG=en_US.utf8 PACKAGECLOUD_TOKEN=<% $.token %> package_cloud push StackStorm/<% $.repo %>/ubuntu/trusty /tmp/st2-up/ubuntu/trusty/<% $.deb %>"
```

Comment:
is `package_cloud` guaranteed to be present on the box running this command?

You may have to update the corresponding profile in https://github.com/StackStorm/st2puppet.